main.tf

```
@@ -1,24 +1,47 @@
 resource "aws_cloudwatch_event_rule" "this" {
-  for_each = { for item in var.cloudwatch_rule : item.event_rule.name => item }
 
-  name          = each.value.event_rule.name
-  description   = each.value.event_rule.description
-  event_pattern = each.value.event_rule.event_pattern
+  name          = var.cloudwatch_rule.event_rule.name
+  description   = var.cloudwatch_rule.event_rule.description
+  event_pattern = var.cloudwatch_rule.event_rule.event_pattern
 }
 
-
 resource "aws_cloudwatch_event_target" "this" {
-  for_each = { for item in var.cloudwatch_rule : item.event_target.event_name => item }
 
-  arn       = each.value.event_target.event_target_arn
-  rule      = each.value.event_target.event_name
-  target_id = each.value.event_target.event_target_id
-  role_arn  = each.value.event_target.role_arn
+  for_each = tomap({
+    for target in local.rule_target_pairings : "${target.k}" => target
+  })
```

With the suggested variable definition, this could just be:

```suggestion
  for_each = { for target in var.event_rule.event_targets : target.name => target }
```